docs(blockchain-scanner): Update documentation for the new `zebra-scanner` binary (#8675)
ZcashFoundation/zebra: book/src/user/shielded-scan.md

# Zebra Shielded Scanning

The `zebra-scanner` binary is a standalone application that utilizes Zebra libraries to scan for transactions associated with specific Sapling viewing keys. It stores the discovered transactions and scanning progress data in a RocksDB database.

> For this application to function, it requires a `zebrad` node to be running concurrently with the scanner application.

Reviewer: Nitpick/Optional: Could we also mention that it needs access to Zebra's state cache?

```suggestion
For this application to function, it requires access to a Zebra node's RPC server and state cache.
```

 
For now, we only support Sapling, and only store transaction IDs in the scanner results database.

> Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728) and [#8573](https://github.com/ZcashFoundation/zebra/issues/8573).

Reviewer: Optional: It looks like #8573 is almost done if we want to omit it here.
```suggestion
Ongoing development is tracked in issue [#7728](https://github.com/ZcashFoundation/zebra/issues/7728).
```

## Important Security Warning

Zebra's shielded scanning feature has known security issues. It is for experimental use only.

> Do not use regular or sensitive viewing keys with Zebra's experimental scanning feature. Do not use this feature on a shared machine. We suggest generating new keys for experimental use or publicly known keys.

Reviewer: ```suggestion
Do not use regular or sensitive viewing keys with Zebra's experimental scanning feature. Do not use this feature on a shared machine. We suggest generating new keys for experimental use or using publicly known keys.
```

## Build & Install

Use [Zebra 1.9.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.9.0) or greater, or the `main` branch to get the latest features.

You can also use Rust's `cargo` to install `zebra-scanner` from the latest release Zebra repository:

```bash

> cargo install --locked --git https://github.com/ZcashFoundation/zebra zebra-scan

Reviewer: This probably also installs `zebrad` (as `zebrad-for-scanner`), we may want to hide that behind a feature later.

```

The scanner binary will be at `~/.cargo/bin/zebra-scanner`, which should be in your `PATH`.

## Arguments

Retrieve the binary arguments with:

```bash
zebra-scanner --help
```

## Scanning the Block Chain

> Before starting, ensure a `zebrad` node is running locally with the RPC endpoint open. Refer to the [lightwalletd zebrad setup](https://zebra.zfnd.org/user/lightwalletd.html#configure-zebra-for-lightwalletd) or [zebrad mining setup](https://zebra.zfnd.org/user/mining.html#configure-zebra-for-mining) for instructions.

Reviewer: Optional: We could be more specific about what's needed here too, but "is running locally" also seems fine, hopefully this won't be a constraint in the future anyway.

To initiate the scanning process, you need the following:

- A zebrad cache state directory. This can be obtained from the running zebrad configuration file, under the `state` section in the `cache_dir` field.
- A key to scan with, optionally including a birthday height, which specifies the starting height for the scanning process for that key.
- A zebrad RPC endpoint address. This can be found in the running zebrad configuration file, under the `rpc` section in the `listen_addr` field.

> Pass the data as arguments, for example:

Reviewer: ```suggestion
Sapling diversifiable/extended full viewing keys strings start with `zxviews` as
described in
[ZIP-32](https://zips.z.cash/zip-0032#sapling-extended-full-viewing-keys).

For example, to scan the block chain with the [public ZECpages viewing
key](https://zecpages.com/boardinfo), use:
```
